hypothesis-python/tests/cover/test_rule_argument_validation.py

```diff
@@ -0,0 +1,55 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Most of this work is copyright (C) 2013-2019 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import absolute_import, division, print_function
+
+import pytest
+
+from hypothesis.errors import InvalidArgument
+from hypothesis.searchstrategy.numbers import WideRangeIntStrategy
+from hypothesis.searchstrategy.strategies import OneOfStrategy, SearchStrategy
+from hypothesis.stateful import validate_rule_arguments
```

Comment:
We tend to prefer that tests go though the public API, so adding an `InvalidRuleMachine` and `InvalidInvariantMachine` to `test_stateful.py` would suffice.  The valid paths will be covered by existing tests.

Reply:
Hey @Zac-HD, 

Thanks for your quick reply and feedback! I have updated my pull request. I am still getting familiar with the hypothesis code base, and if I have overlooked something or you would like me to make any further changes, definitely let me know.